src/error.rs

```diff
@@ -11,11 +11,11 @@ use std::{
     result,
 };
 /// Shorthand for a Result that returns an `OrcaError`.
-pub type Result<T> = result::Result<T, OrcaError>;
+pub(crate) type Result<T> = result::Result<T, OrcaError>;
```

Comment:
Given we will have methods that will actually return this Result<T>, unavailability of this type definition will make the code uncompilable the moment those methods using Result<T> return types are used outside of our crate.